dotcom-rendering/src/model/enhance-ad-placeholders.test.ts

```diff
@@ -73,4 +83,43 @@ describe('Enhancing ad placeholders', () => {
 			}
 		},
 	);
+
+	describe('should not insert an ad placeholder after a non text element', () => {
+		const threeParagraphs = getTestParagraphElements(3);
+
+		const elements = [
+			...threeParagraphs,
+			getImageElement(),
+			...threeParagraphs,
+		];
+
+		const input: Block[] = [
+			{
+				...blockMetaData,
+				elements,
+			},
+		];
+
+		const output = enhanceAdPlaceholders(input);
+		const outputElements = getElementsFromBlocks(output);
+		const outputPlaceholders = outputElements.filter(
+			elementIsAdPlaceholder,
+		) as AdPlaceholderBlockElement[];
```

Comment:
this should be placed in the `elementIsAdPlaceholder` signature:

`const elementIsAdPlaceholder = (element: FEElement): element is AdPlaceholderBlockElement`